#! /bin/sh /usr/share/dpatch/dpatch-run
## umask-fix.dpatch by Bernd Zeimetz <bernd@bzed.de>
##
## DP: bugfix from svn r391, see
## DP: http://code.google.com/p/modwsgi/issues/detail?id=18
## DP: for details

@DPATCH@
diff -urNad mod-wsgi~/mod_wsgi.c mod-wsgi/mod_wsgi.c
--- mod-wsgi~/mod_wsgi.c	2007-07-25 06:14:43.000000000 +0200
+++ mod-wsgi/mod_wsgi.c	2007-07-30 18:46:00.000000000 +0200
@@ -5000,7 +5000,7 @@
                 return "Invalid umask for WSGI daemon process.";
 
             errno = 0;
-            umask = strtol(value, (char **)&value, 7);
+            umask = strtol(value, (char **)&value, 8);
 
             if (*value || errno == ERANGE || umask < 0)
                 return "Invalid umask for WSGI daemon process.";
